alice/middleware/logger.go

```diff
@@ -34,15 +36,16 @@ func NewLogrusLogger(baseEntry *logrus.Entry, logRequests bool) Logger {
 	return &logger{entry: baseEntry, logRequests: logRequests}
 }
 
+// Log is the middleware for injecting a logger into the context that has
+// request specific information
 func (l *logger) Log(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		loggingResponseWriter := &lrw.LoggingResponseWriter{w, http.StatusOK, nil}
-
+		// base fields that get added to each log entry
 		fields := logrus.Fields{
-			logFieldRequestID: GetRequestIDFromContext(r.Context()),
-			logFieldUserID:    GetInsecureUserIDFromContext(r.Context()),
-			logFieldMethod:    r.Method,
-			logFieldPath:      r.URL.String(),
+			logFieldUserID:  GetInsecureUserIDFromContext(r.Context()),
+			logFieldMethod:  r.Method,
+			logFieldPath:    r.URL.String(),
+			logFieldTraceID: r.Header.Get("X-Datadog-Trace-ID"),
```

Comment:
Is there documentation about this header? If so, having the link to it in a comment here would probably help maintainability in the future.